src/clients/c++/library/request_common.h

```diff
@@ -40,61 +41,87 @@
 
 namespace nvidia { namespace inferenceserver { namespace client {
 
-// Timer to record the timestamp for different stages of request
-// handling.
+// Records timestamps for different stages of request handling.
 class RequestTimers {
  public:
-  using ClockType = std::chrono::steady_clock;
-  using TimePoint = ClockType::time_point;
-
-  /// Return the duration between start time point and end timepoint
-  /// in nanosecond.
-  /// \param start The start time point.
-  /// \param end The end time point.
-  /// \return Duration in nanosecond.
-  static uint64_t Duration(const TimePoint& start, const TimePoint& end)
-  {
-    return std::chrono::duration_cast<std::chrono::nanoseconds>(end - start)
-        .count();
-  }
-
-  /// The kind of the timer.
-  enum Kind {
+  /// Timestamp kinds.
+  enum class Kind {
     /// The start of request handling.
     REQUEST_START,
+
     /// The end of request handling.
     REQUEST_END,
-    /// The start of sending request bytes to the server (i.e. first byte).
+
+    /// The start of sending request bytes to the server (i.e. first
+    /// byte).
     SEND_START,
-    /// The end of sending request bytes to the server (i.e. last byte).
+
+    /// The end of sending request bytes to the server (i.e. last
+    /// byte).
     SEND_END,
+
     /// The start of receiving response bytes from the server
     /// (i.e. first byte).
-    RECEIVE_START,
-    /// The end of receiving response bytes from the server
-    /// (i.e. last byte).
-    RECEIVE_END
+    RECV_START,
+
+    /// The end of receiving response bytes from the server (i.e. last
+    /// byte).
+    RECV_END,
+
+    COUNT__
   };
 
   /// Construct a timer with zero-ed timestamps.
-  RequestTimers() { Reset(); }
+  RequestTimers() : timestamps_((size_t)Kind::COUNT__) { Reset(); }
 
   /// Reset all timestamp values to zero. Must be called before
   /// re-using the timer.
-  /// \return Error object indicating success or failure.
-  Error Reset();
-
-  /// Record the current timestamp for a request stage.
-  /// \param kind The Kind of the timestamp.
-  /// \return Error object indicating success or failure.
-  Error Record(Kind kind);
-
-  TimePoint request_start_;
-  TimePoint request_end_;
-  TimePoint send_start_;
-  TimePoint send_end_;
-  TimePoint receive_start_;
-  TimePoint receive_end_;
+  void Reset()
+  {
+    memset(&timestamps_[0], 0, sizeof(uint64_t) * timestamps_.size());
+  }
+
+  /// Get the timestamp, in nanoseconds, for a kind.
+  /// \param kind The timestamp kind.
+  /// \return The timestamp in nanoseconds.
+  uint64_t Timestamp(Kind kind) const { return timestamps_[(size_t)kind]; }
+
+  /// Set a timestamp to the current time, in nanoseconds.
+  /// \param kind The timestamp kind.
+  /// \return The timestamp in nanoseconds.
+  uint64_t CaptureTimestamp(Kind kind)
+  {
+    struct timespec spec;
+    clock_gettime(CLOCK_MONOTONIC, &spec);
```

Comment:
We used `<chrono>` previously because `timespec` etc. are POSIX functionalities, which is not portable to Windows. Need to make sure client library builds on Windows...